app/presenters/shelf_list_presenter.rb

```diff
@@ -16,7 +16,8 @@ class ShelfListPresenter
   def initialize(params = {})
     @starting = params[:starting]
     @ending = params[:ending]
-    @nearby = params[:nearby]
+    @label = params[:nearby]
+    @nearby = Shelvit.normalize(params[:nearby]) || params[:nearby]
```

Comment:
I'd probably make this a method:

``` ruby
def nearby
  @nearby ||= Shelvit.normalize(params[:nearby]) || params[:nearby]
end
```

Reply:
See my earlier comment. This might ought to be a different method. The param and the method could be returning different things which might be confusing.

Reply:
What if we keep `@nearby = params[:nearby]` and then make a `shelf_key` method?

```ruby
def shelf_key
  Shelvit.normalize(nearby) || nearby
end
```

Reply:
works for me! Memoize it, if gets called more than once.